pkg/storage/buckets/store.go

```diff
@@ -88,15 +68,29 @@ func (b *BucketStore) Get(ctx context.Context, bucket string) (meta Meta, err er
 		return Meta{}, storj.ErrNoBucket.New("")
 	}
 
-	objMeta, err := b.store.Meta(ctx, bucket)
+	bb, objectPath, segmentIndex, err := segments.SplitPathFragments(bucket)
 	if err != nil {
-		if storage.ErrKeyNotFound.Has(err) {
-			err = storj.ErrBucketNotFound.Wrap(err)
-		}
 		return Meta{}, err
 	}
 
-	return convertMeta(objMeta)
+	pointer, err := b.metainfo.SegmentInfo(ctx, bb, objectPath, segmentIndex)
+	if err != nil {
+		return Meta{}, err
+	}
+
+	// metadata conversion function call: pointer -> segment -> stream -> object -> bucket
```

Comment:
This conversion from pointer to bucket `Meta` object would be simplified if there were a metainfo call available like `b.metainfo.BucketInfo()`, but currently that doesn't exist.  However, it does seem like that change is captured in the metainfo refactor. See design doc for details: https://github.com/storj/storj/blob/e76ecf47d8609fa1d873f9f3e00134f7853b85ad/docs/design/metainfo-rpc.md

In the meantime, since this refactor is not implemented yet, maybe what you have here is good.  Maybe add a TODO comment about once refactor is done, we can replace this conversion. Or not. up to you

Reply:
good idea! i'll make a temp method to do this